File: Wirtualny świat/java/src/organizm/roslina.java

package organizm;
import java.util.Random;
import swiat.swiat;

public abstract class roslina extends organizm {
    roslina(swiat WielkiSwiat, int x, int y)
    {
        super(WielkiSwiat, x, y);
    }
        public void kolizja(organizm o,organizm o2)
    {
         }
        
    @Override
    public void akcja(swiat ref) {
	Random r = new Random();
        if (r.nextBoolean()) { // 50% szansy na rozmnorzenie rosliny
                int ra = r.nextInt(4)+1; //starts at 0, so add 1
		int x = this.x;
		int y = this.y;

		   boolean[] co_sprawdzilismy = { false,false,false,false };

		if (this.x == 0 && this.y == 0) //lewa gora
		{
			co_sprawdzilismy[0] = true; //gora
			co_sprawdzilismy[2] = true; //lewo
		}
		else if (this.x == N - 1 && this.y == 0)  //prawa gora
		{
			co_sprawdzilismy[0] = true; //gora
			co_sprawdzilismy[3] = true; // prawo
		}
		else if (this.x == 0 && this.y == N - 1)  //lewy dol
		{
			co_sprawdzilismy[1] = true; //dol
			co_sprawdzilismy[2] = true; //lewo
		}
		else if (this.x == N - 1 && this.y == N - 1) //prawy dol
		{
			co_sprawdzilismy[1] = true; //dol
			co_sprawdzilismy[3] = true; // prawo
		}
		else if (this.x == 0 && this.y > 0 && this.y < N) //krawędź lewo
		{
			co_sprawdzilismy[2] = true; //lewo
		}
		else if (this.x == N - 1 && this.y > 0 && this.y < N) //krawędź prawo
		{
			co_sprawdzilismy[3] = true; // prawo		
		}
		else if (this.y == 0 && this.x > 0 && this.x < N) //krawędź gora
		{
			co_sprawdzilismy[0] = true; //gora
		}
		else if (this.y == N - 1 && this.x > 0 && this.x < N) //krawędź dol
		{
			co_sprawdzilismy[1] = true; //dol
		}

		while (co_sprawdzilismy[ra-1] != false) { ra = r.nextInt(4)+1; };//losujemy r tak dlugo az trafimy na dozwolone pole
		 // czy kolizja?

		switch (ra)
		{
		case 1: --y; break;
		case 2: ++y; break;
		case 3: --x; break;
		case 4: ++x; break;
		}

		
		if (!ref.czyZajete(x, y)) {
			if (ref.organizm_na_polu(x, y) == null) {
				this.nowy(ref, x, y);
                                 System.out.println("dodano " + this.rysowanie() + " na polu: " + (x+1) + ":" + (y+1));
			}
		}
	}
}
}

File: Wirtualny świat/C++/ConsoleApplication18/roslina.cpp
#include "stdafx.h"
#include "roslina.h"


roslina::roslina(swiat& ref, int x, int y) :organizm(ref, x, y)
{
}
void roslina::akcja(swiat& ref) {
	if((rand() % 2) != 0)	{ // 50% szansy na rozmnorzenie rosliny
	int r = (rand() % 4) + 1;
	int x = this->x;
	int y = this->y;
	
	bool co_sprawdzilismy[4] = { 0,0,0,0 };  //gora dol lewo prawo -- gdzie nie mozna isc

	if (this->x == 0 && this->y == 0) //lewa gora
	{  
		co_sprawdzilismy[0] = 1; //gora
		co_sprawdzilismy[2] = 1; //lewo
	}
	else if (this->x == N - 1 && this->y == 0)  //prawa gora
	{
		co_sprawdzilismy[0] = 1; //gora
		co_sprawdzilismy[3] = 1; // prawo
	}
	else if (this->x == 0 && this->y == N - 1)  //lewy dol
	{
		co_sprawdzilismy[1] = 1; //dol
		co_sprawdzilismy[2] = 1; //lewo
	}
	else if (this->x == N - 1 && this->y == N - 1) //prawy dol
	{
		co_sprawdzilismy[1] = 1; //dol
		co_sprawdzilismy[3] = 1; // prawo
	}
	else if (this->x == 0 && this->y > 0 && this->y < N) //krawędź lewo
	{
		co_sprawdzilismy[2] = 1; //lewo
	}
	else if (this->x == N - 1 && this->y > 0 && this->y < N) //krawędź prawo
	{
		co_sprawdzilismy[3] = 1; // prawo		
	}
	else if (this->y == 0 && this->x > 0 && this->x < N) //krawędź gora
	{
		co_sprawdzilismy[0] = 1; //gora
	}
	else if (this->y == N - 1 && this->x > 0 && this->x < N) //krawędź dol
	{
		co_sprawdzilismy[1] = 1; //dol
	}



	for (; co_sprawdzilismy[r-1] != 0; r = (rand() % 4) + 1) {}; //losujemy r tak dlugo az trafimy na dozwolone pole
	
	// czy kolizja?

	switch (r)
	{
	case 1: --y; break;
	case 2: ++y; break;
	case 3: --x; break;
	case 4: ++x; break;
	}

	
	if (!ref.czyZajete(x, y)) 
	{
		if (ref.organizm_na_polu(x, y) == NULL) 
		{
		this->nowe(ref, x, y);
		}
	}
 }
}
void roslina::kolizja(swiat& ref,organizm* ten, organizm* tamten)   //nie wystepuje
{
	if (ten->sila_organizmu() > tamten->sila_organizmu()) {
		tamten->ustaw_czy_zyje(false);
		cout << "roslina zabila zwierze - BLAD\n";
	}
	else if (ten->sila_organizmu() < tamten->sila_organizmu()) {
		ten->ustaw_czy_zyje(false);
		cout << "roslina zginela - BLAD\n";
	}
}
roslina::~roslina()
{
}
